packages/core/src/mixins/with-events.ts

```diff
@@ -0,0 +1,49 @@
+import mitt from 'mitt';
+import { CartoError } from '../error/carto-error';
+
+export abstract class WithEvents {
```

Comment:
The name sounds a bit weird for me (I don't expect an event emitter), I would prefer using `Emitter` like `CARTOEmitter` or even `EventEmitter` like the node one

Reply:
Not strong opinion on this, but I would leave it as it (eg Mapbox has Evented, which is pretty similar concept)

Reply:
Yeah, the thing is that it is not an event emitter per se, but a class that implements a mechanism to handle and manage events.

The initial approach with multiple inheritance was to extend the class like this:
```js
WithEvents(Dataview)
```

But that approach was not needed because all the classes that we needed to extend with this one didn't inherit from another class. I decided to leave the name though as it didn't look bad with:

```js
export class DataView extends WithEvents {}
```